scipy/stats/morestats.py

```diff
@@ -2721,50 +2722,109 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False):
+        If `alternative` is two-sided, the sum of the ranks of the
+        differences above or below zero, whichever is smaller.
+        Otherwise the sum of the ranks of the differences above zero.
     pvalue : float
-        The two-sided p-value for the test.
+        The p-value for the test depending on `alternative`.
+
+    See Also
+    --------
+    kruskal, mannwhitneyu
 
     Notes
     -----
-    Because the normal approximation is used for the calculations, the
-    samples used should be large.  A typical rule is to require that
-    n > 20.
+    The test has been introduced in [4]_. Given n independent samples
+    (xi, yi) from a bivariate distribution, compute the differences
+    di = xi - yi. One assumption of the test is that the differences
+    are symmetric, see [2]_.
+    The two-sided test has the null hypothesis that the median of the
+    differences is zero against the alternative that it is different from
+    zero. The one-sided test has the null that the median is positive against
+    the alternative that the it is negative (``alternative == 'less'``),
+    or vice versa (``alternative == 'greater.'``).
+
+    The test uses a normal approximation to derive the p-value. A typical rule
+    is to require that n > 20. For smaller n, exact tables can be used to find
+    critical values.
 
     References
     ----------
     .. [1] https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test
+    .. [2] Conover, W.J., Practical Nonparametric Statistics, 1971.
+    .. [3] Pratt, J.W., Remarks on Zeros and Ties in the Wilcoxon Signed
+       Rank Procedures, Journal of the American Statistical Association,
+       Vol. 54, 1959, pp. 655-667.
+    .. [4] Wilcoxon, F., Individual Comparisons by Ranking Methods,
+       Biometrics Bulletin, Vol. 1, 1945, pp. 80-83.
+
+    Examples
+    --------
+    In [4]_, the differences in height between cross- and self-fertilized
+    corn plants is given as follows:
+
+    >>> d = [6, 8, 14, 16, 23, 24, 28, 29, 41, -48, 49, 56, 60, -67, 75]
+
+    Cross-fertilized plants appear to be be higher. To test the null that
+    there is no height difference, we can apply the two-sided test:
+
+    >>> from scipy.stats import wilcoxon
+    >>> w, p = wilcoxon(d)
+    >>> w, p
+    (24.0, 0.04088813291185591)
+
+    Hence, we would reject the null at a confidence level of 5%, concluding
+    that there is a difference in height between the groups. To confirm
+    that the median of the differences can be assumed to be positive, we use:
+
+    >>> w, p = wilcoxon(d, alternative='greater')
+    >>> w, p
+    (96.0, 0.020444066455927955)
+
+    This shows that the null that the median is negative can be rejected at
+    a confidence level of 5% in favor of the alternative that the median is
+    greater than zero. The p-value based on the approximation is within the
+    range of 0.019 and 0.054 given in [2]_.
+    Note that the statistic changed to 96 in the one-sided case (the sum
```

Comment:
"one-sided greater case"? To distinguish from "one sided lesser" case where the statistic wouldn't be 96.

Reply:
> Or would it still be 96, but the cdf is used instead of the sf to calculate the p-value?

yes